packages/client/hmi-client/src/components/models/tera-model.vue

```diff
@@ -866,20 +913,89 @@ async function updateModelName() {
 	if (model.value) {
 		const modelClone = cloneDeep(model.value);
 		modelClone.name = newModelName.value;
-		updateModel(modelClone);
-		isRenamingModel.value = false;
+		await updateModel(modelClone);
 		model.value = await getModel(props.assetId);
 		// FIXME: Names aren't updated in sidebar
 	}
+	isRenamingModel.value = false;
 }
 
-// Toggle rows to become editable, display extractions to choose from.
-function editSection(event: Event) {
+// Toggle rows to become editable
+function editRow(event: Event) {
 	if (!event?.target) return;
-	const section = (event.target as HTMLElement).closest('.datatable section');
-	if (!section) return;
-	isSectionEditable.value =
-		isSectionEditable.value === section.className ? null : section.className;
+	const row = (event.target as HTMLElement).closest('.datatable section');
+	if (!row) return;
+	isRowEditable.value = isRowEditable.value === row.className ? null : row.className;
+}
+
+async function confirmEdit() {
+	if (model.value && transientTableValue.value) {
+		const { tableType, idx, updateProperty } = transientTableValue.value;
+		const modelClone = cloneDeep(model.value);
+
+		switch (tableType) {
+			case 'parameters':
+				if (model.value.semantics?.ode.parameters) {
+					Object.entries(updateProperty).forEach(([key, value]) => {
+						modelClone.semantics!.ode.parameters![idx][key] = value;
+
+						if (key === 'id') {
+							const ode = model.value!.semantics!.ode;
+							// if we update parameter id we also need to update rate expression and expression_mathml
+							const updatedRates = ode.rates.map((rate) => ({
+								...rate,
+								expression: replaceValuesInExpression(
+									rate.expression,
+									ode.parameters![idx][key],
+									value as string
+								),
+								expression_mathml: rate.expression_mathml
+									? replaceValuesInMathML(
+											rate.expression_mathml,
+											ode.parameters![idx][key],
+											value as string
+									  )
+									: undefined
+							}));
+							modelClone.semantics!.ode.rates = updatedRates;
+						}
+					});
+				}
+				break;
+			case 'states':
+				Object.entries(updateProperty).forEach(([key, value]) => {
+					if (key !== 'unit') {
+						// TODO: remove this condition when we proper editing of unit
+						modelClone.model.states[idx][key] = value;
+					}
+					// TODO: update all of the properties affected by state id
+				});
+				break;
+			default:
+				logger.info(`${tableType} not recognized`);
+		}
```

Comment:
Might be worthwhile to see if the update logic (e.g. update parameter) can be moved into petrinet-service alongside other update functions. That way the logic isn't scattered across the code base.

Reply:
created some helper functions in `petrinet-service.ts`, let me know if it looks okay to you